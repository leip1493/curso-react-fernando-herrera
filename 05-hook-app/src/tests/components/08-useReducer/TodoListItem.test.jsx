import { shallow } from "enzyme";
import React from "react";
import { todos } from "../../fixtures/todos";
const {
  TodoListItem,
} = require("../../../components/08-useReducer/TodoListItem");

describe("<TodoListItem />", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={todos[0]}
      index={0}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );
  test("Debe renderizar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe llamar la funcion handleDelete", () => {
    wrapper.find("button").simulate("click");

    expect(handleDelete).toHaveBeenCalledWith(todos[0].id);
  });

  test("Debe llamar la funcion handleToggle", () => {
    wrapper.find("p").simulate("click");

    expect(handleToggle).toHaveBeenCalledWith(todos[0].id);
  });

  test("Debe mostrar el texto correctamente", () => {
    const p = wrapper.find("p");
    console.log(p.text());
    expect(p.text().trim()).toBe(`${0 + 1}. ${todos[0].description}`);
  });

  test("Debe tener la clase complete si TODO.done == true", () => {
    const todo = todos[0];
    todo.done = true;

    const wrapper = shallow(<TodoListItem todo={todo} />);

    expect(wrapper.find("p").hasClass("complete")).toBeTruthy()
  });
});
