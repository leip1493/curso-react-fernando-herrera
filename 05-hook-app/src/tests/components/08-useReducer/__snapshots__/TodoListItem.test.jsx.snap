// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> Debe renderizar correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1. undefined
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
